tools/shared.py

```diff
@@ -188,7 +198,8 @@ def get_finished_process():
             logger.info(out)
           if err:
             logger.error(err)
-          raise Exception('Subprocess %d/%d failed with return code %d! (cmdline: %s)' % (idx + 1, len(commands), finished_process.returncode, shlex_join(commands[idx])))
+
+          raise Exception('Subprocess %d/%d failed (%s)! (cmdline: %s)' % (idx + 1, len(commands), returncode_to_str(finished_process.returncode), shlex_join(commands[idx])))
```

Comment:
This isn't tested, and I'm not sure how to test it in a reasonable way. But I'd recommend at least testing it manually locally before landing (say adding a throw in the system libs code, and building a system lib). Just to rule out a typo on this line that ends up masking the proper reported error.

Reply:
Done.